hydra/callback/mlflow.py

```diff
@@ -0,0 +1,24 @@
+import logging
+import os
+
+from omegaconf import DictConfig
+
+from hydra.callback.callback import Callback
+import mlflow
+
+log = logging.getLogger(__name__)
+
+
+class MLflow(Callback):
+    def __init__(self, tracking_uri: str):
+        self.tracking_uri = tracking_uri
+
+    def _set_tracking_uri(self):
+        log.info(f"Setting mlflow tracking URI to be {self.tracking_uri}")
+        mlflow.set_tracking_uri(self.tracking_uri)
```

Comment:
Is this an idempotent function? I think atleast some ml lifecycle plugins will complain if some metadata is set several times within the same process

Reply:
Good point! :) 
I will definitely test it more thoroughly once we finalized on the Callbacks interface and config. 
Also, this function should only be called once (either by `pre_run` or `pre_sweep`) during a Hydra app's lifecycle.